### DISCUSSION TITLE
How to set default sort for DataGrid

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#13417] [Q&A] [answered]
I can't figure out how to set the default sort column and direction in the DataGrid.
The DataGridCollectionView looks powerful, but I can't set that as the ItemsSource without breaking data binding of the columns.

Any suggestions?

**Comment by thevortexcloud** [comment, marked as answer]
There is an example here on how to use `DataGridCollectionView` to do what you want.

https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/DataGridPage.xaml.cs
.

**Reply by AArnott** [reply]
Thanks for the super-fast reply. That appears to use the DataGridCollectionView as the ItemsSource as I was trying to do, but it sets that programmatically instead of in XAML. 
I see that works. Thank you.
It seems kinda odd that I can't set it in the XAML directly without breaking data binding in the columns. The DataGridCollectionView isn't a generic collection so I suppose the designer can't tell what the columns to be bound would be. So at the moment, I bind against the `ObservableCollection<T>` in the XAML, and then in the code-behind I override that to be the `DataGridCollectionView`, and everything works.

If there's a better way to do that, please advise. I appreciate the pointer.

**Reply by thevortexcloud** [reply]
I personally have bound to a `DataGridCollectionView` in a view model before via XAML without issue before, so I am not sure what's wrong with your bindings. But I am not a big fan of doing that anyway, Since that:

1. Directly couples the view model to Avalonia which make reusing code harder
2. Means the view model is now worried about how things are presented, which is generally a view concern

But it should still work as far as I am aware. With that said I also have not spent a huge amount of time messing with datagrids so I can't really suggest a better way.

**Reply by timunie** [reply]
I prefer to use DynamicData for custom sorting / filtering and disable the sorting of DataGrid. It gives me more flexibility but also adds some more code to write.

**Reply by thevortexcloud** [reply]
Dynamic Data does have some really nice features. But it's also significantly more complicated to use. Which can make it fairly painful to use for small stuff.

**Reply by peterGdot** [reply]
Really good work! Is there a trick to using the AsyncVirtualisingCollection class? When I use it, my grid is left blank.
